### ISSUE TITLE
Error with local filesystem packages

### ISSUE THREAD

**Issue by staltz** [ds300/patch-package#3]
**Summary:**

I have a package.json where some packages are installed from local filesystem, not npm, such as 

```
"process-scarecrow": "file:./process-scarecrow",
```

But I'm patching something else, `patch-package ssb-keys`. (I had initially installed all packages with yarn, just FYI). Then I get an error which apparently aborts the patching process so I don't get the file `patches/ssb-keys:1.2.3.patch`.

**Expected:**

Patch file created and the module is patched as usual.

**Actual:**

Error when running `patch-package ssb-keys`

```

***Warning***
patch-package currently only works after `yarn remove` when you install a
project-local copy of yarn.
See https://github.com/ds300/patch-package#why-patch-yarn for details

error "/private/var/folders/4k/xjj8sw652m7046nc2006k8fc0000gn/T/tmp-10745p4f76427JQ7b/process-scarecrow" doesn't exist.
{ Error: Command failed: yarn
error "/private/var/folders/4k/xjj8sw652m7046nc2006k8fc0000gn/T/tmp-10745p4f76427JQ7b/process-scarecrow" doesn't exist.

    at checkExecSyncError (child_process.js:481:13)
    at Object.execSync (child_process.js:521:13)
    at tmpExec (/Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/makePatch.js:35:63)
    at Object.makePatch [as default] (/Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/makePatch.js:39:9)
    at /Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/index.js:21:32
    at Array.forEach (native)
    at Object.<anonymous> (/Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/index.js:20:22)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
  error: null,
  cmd: 'yarn',
  file: '/bin/sh',
  args: [ '/bin/sh', '-c', 'yarn' ],
  options:
   { cwd: '/var/folders/4k/xjj8sw652m7046nc2006k8fc0000gn/T/tmp-10745p4f76427JQ7b',
     shell: true,
     file: '/bin/sh',
     args: [ '/bin/sh', '-c', 'yarn' ],
     envPairs: [],
     stdio: [ [Object], [Object], [Object] ] },
  envPairs: [],
  stderr: <Buffer 65 72 72 6f 72 20 22 2f 70 72 69 76 61 74 65 2f 76 61 72 2f 66 6f 6c 64 65 72 73 2f 34 6b 2f 78 6a 6a 38 73 77 36 35 32 6d 37 30 34 36 6e 63 32 30 30 ... >,
  stdout: <Buffer 79 61 72 6e 20 69 6e 73 74 61 6c 6c 20 76 30 2e 32 34 2e 35 0a 5b 31 2f 34 5d 20 52 65 73 6f 6c 76 69 6e 67 20 70 61 63 6b 61 67 65 73 2e 2e 2e 0a 69 ... >,
  pid: 10750,
  output:
   [ null,
     <Buffer 79 61 72 6e 20 69 6e 73 74 61 6c 6c 20 76 30 2e 32 34 2e 35 0a 5b 31 2f 34 5d 20 52 65 73 6f 6c 76 69 6e 67 20 70 61 63 6b 61 67 65 73 2e 2e 2e 0a 69 ... >,
     <Buffer 65 72 72 6f 72 20 22 2f 70 72 69 76 61 74 65 2f 76 61 72 2f 66 6f 6c 64 65 72 73 2f 34 6b 2f 78 6a 6a 38 73 77 36 35 32 6d 37 30 34 36 6e 63 32 30 30 ... > ],
  signal: null,
  status: 1 }
/Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/makePatch.js:64
        throw e;
        ^

Error: Command failed: yarn
error "/private/var/folders/4k/xjj8sw652m7046nc2006k8fc0000gn/T/tmp-10745p4f76427JQ7b/process-scarecrow" doesn't exist.

    at checkExecSyncError (child_process.js:481:13)
    at Object.execSync (child_process.js:521:13)
    at tmpExec (/Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/makePatch.js:35:63)
    at Object.makePatch [as default] (/Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/makePatch.js:39:9)
    at /Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/index.js:21:32
    at Array.forEach (native)
    at Object.<anonymous> (/Users/staltz/.nvm/versions/node/v6.10.3/lib/node_modules/patch-package/dist/index.js:20:22)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
```

**More:**

node: v6.10.3
yarn: v0.24.5
patch-package: 2.1.1

**Comment by ds300** [comment]
Hmm yeah that makes sense. It could replace relative `file:` deps in the package.json with absolute paths to fix this. Will have to wait until next week though as I'm off to fututrip tomorrow.

**Comment by ds300** [comment]
Forgot to comment to say that this should be fixed now in  `patch-package@3.0.0`

**Comment by staltz** [comment]
Thanks @ds300 ! I haven't tested it yet, I'm on mobile. But will test
